apps/studio/src/lib/data/jsonViewer.ts

```diff
@@ -137,3 +138,37 @@ export function eachPaths(
     }
   }
 }
+
+/** Mutate the row data to replace JSON columns with parsed JSON objects */
+export function parseRowDataForJsonViewer(data: Record<string, any>, tableColumns: { field: string, dataType: string }[]) {
+  tableColumns.forEach(column => {
+    const columnValue = data[column.field]
+
+    // Check if the column is a JSON column
+    let isJsonColumn = String(column.dataType).toUpperCase() === 'JSON' || String(column.dataType).toUpperCase() === 'JSONB'
+
+    // If the column is not a JSON column, check if it is a JSON string
+    if (!isJsonColumn) {
+      const isColumnHasStringAndNotEmpty = typeof columnValue === 'string' && columnValue.trim() !== ''
+
+      if (isColumnHasStringAndNotEmpty) {
+        const isJsonObjectString = columnValue.startsWith('{') && columnValue.endsWith('}')
```

Comment:
You'll need to trim whitespace here too -- spaces at the end or start of this string will break the detection.

This should have a unit test